### PR TITLE
CTED-2305 meet the students block settings and config

### PR DESCRIPTION
**Pull request by loyisongq**

### CODE REVIEW

File: block_meet_the_students.php [hunk 1]
```diff
@@ -123,6 +122,19 @@ protected function render_user_pictures($context) {
...
+        $companyposition = (isset($this->config->displayclassmatescompanyposition) ? $this->config->displayclassmatescompanyposition : $config->displayclassmatescompanyposition) + 1;
+        $linkedin = (isset($this->config->displayclassmateslinkedin) ? $this->config->displayclassmateslinkedin : $config->displayclassmateslinkedin) + 1;
+        $skype = (isset($this->config->displayclassmatesskype) ? $this->config->displayclassmatesskype : $config->displayclassmatesskype) + 1;
+        $calendly = (isset($this->config->displayclassmatescalendlyid) ? $this->config->displayclassmatescalendlyid : $config->displayclassmatescalendlyid) + 1;
+        $about = (isset($this->config->displayclassmatesabout) ? $this->config->displayclassmatesabout : $config->displayclassmatesabout);
+        $othercourses = (isset($this->config->displayclassmatesprogrammesenrolled) ? $this->config->displayclassmatesprogrammesenrolled : $config->displayclassmatesprogrammesenrolled) + 1;
+        $studentnumber = (isset($this->config->displayclassmatesstudentnumber) ? $this->config->displayclassmatesstudentnumber : $config->displayclassmatesstudentnumber) + 1;
```
Comment (JJSwanevelder) [inline block_meet_the_students.php:135]: @loyisongq - why is there a check for this->config vs config? Is there a scenario where this->config will be different from config?

File: block_meet_the_students.php [hunk 2]
```diff
@@ -161,12 +173,135 @@ protected function render_user_pictures($context) {
...
+                                 </div>
+                               </div>';
+                    $html .= '</div>
+                            </div>';
+
+
+                } else {
+                    $html .= html_writer::start_tag('div', array('class' => 'user_icon', 'style' => $style));
+                    $html .= $OUTPUT->user_picture($users[$i], array('size' => 100, 'class' => 'user_picture'));
+                    $html .= html_writer::end_tag('div');
```
Comment (JJSwanevelder) [inline block_meet_the_students.php:299]: @loyisongq what is the motivation for using the html_writer vs templates?

File: course.php [hunk 3]
```diff
@@ -15,6 +15,12 @@
 // Set up page parameters.
 $PAGE->set_course($course);
 $PAGE->requires->css('/blocks/meet_the_students/styles.css');
+$PAGE->requires->jquery();
+$PAGE->requires->js_call_amd('core/custom_interaction_events');
+$PAGE->requires->js_call_amd('core_message/message_user_button', 'send', array('#message-user-button'));
+$PAGE->requires->js_call_amd('core_message/toggle_contact_button', 'enhance', array('#toggle-contact-button'));
```
[the remark this answers is not in the document]
Comment (loyisongq) [reply course.php:21]: will remove these, the js wasn't attaching correctly to the buttons on the block, will look at another way of displaying the drawer and retrieving correct data ie messages between the users

**Review by JJSwanevelder — APPROVED** [review]
👍

**Pull request merged**